### ISSUE TITLE
Is there a maximum of 999 images?

### ISSUE THREAD

**Issue by lambcutlet** [xiphiasphotography/plogger_php7#5]
Hi, I've used plogger for a very long time but just recently I was forced to upgrade and thankfully found this php 7 version.
I have encountered an issue as there seems to be a 999 limit to the number of images that can be uploaded.
I have one collection, 10 albums and 999 images.
![image](https://user-images.githubusercontent.com/25547085/134800777-1a42dc16-af95-4a34-942b-353db7b5127f.png)
I have loads more images to add. 
Do you know of any limitations?

**Comment by lambcutlet** [comment]
I found this 
`$max_limit = 1000; // Hard-coded max limit, no matter what you do, you cannot
	// go beyond this limit for number of pictures in thumbnails, slideshow etc.`
in plog-functions.php

Can this number be safely increased?

**Comment by xiphiasphotography** [comment]
Hi,

No limitation as far as I know. I'm running 2 large plogger albums.
*Gallery Stats:* You have *16 *collections, which contain *150 *albums and
*8788 *images.
*Gallery Stats*: You have 11 collections, which contain 323 albums and
12448 images.

When I have some time this week I'll do some tests with the dev version I
have running.

**Comment by xiphiasphotography** [comment]
Hi,

No need to increase. This number is the max amount of images to show on 1
page, of max albums on one page. So can have 10000 images in one album but
will be at least 10 pages with 1000 images each. Something I wouldn't
recommend. Most of the time I use 24 images, good for responsive design
showing 4, 3 or 2 images on a row.

There is no limit on how many images can be saved in the database, this is
the max amount of database records that can be retrieved at one time.

You can see where the max_limit is used a few lines below, and it for the
SQL query to retrieve items.

Friendly regards,

**Comment by lambcutlet** [comment]
Hi, 
thanks for replying
So I have 1 collection 10 Albums 999 images.
The background, I have added the next lot of images to the 'uploads' folder 279 to be exact then deselect 7 images as they are partially corrupt. Then choosing an existing Album (one created previously, I can add these) I select import and each time it fails.
After trying numerous times I thought i'd try uploading a single image and it worked, then another so my total count now is 1001. So the issue is not a Max Limit of 1000.
I then proceeded to upload a batch of 50 and it worked and then finally the remainder of 214 images, also worked.
Next album has 160 images, uploaded OK.
Next album has 425 and it failed. I'm starting to think there is a limit on the number of images that can be uploaded in one batch.
250 fails, 249 fails, 240 worked.
Is there possibly a limit to the number of images that can be uploaded in a single batch?
A number between 240 & 249 perhaps?

**Comment by xiphiasphotography** [comment]
I did a similar test yesterday and noticed the same thing. I never uploaded
more then 100 at a time before so never noticed it. I wonder if this is an
existing issue?

I usually add folders in the upload folder with the name of the album and
add the images to each folder. Added bonus is that the images are added to
the album without selecting an album.

Will have to look into the unload restriction limit.
